monitoring.go

```diff
@@ -270,47 +270,23 @@ func (sc *snowflakeConn) waitForCompletedQueryResultResp(
 
 	if ! response.Success {
 		logEverything(ctx, qid, response, startTime)
-		_, statusErr := sc.checkQueryStatus(ctx, qid)
-		logger.WithContext(ctx).Errorf("failed queryId: %v, statusErr: %v", qid, statusErr)
-		retryResponse, retryErr := sc.rest.getAsyncOrStatus(WithReportAsyncError(ctx), url, headers, sc.rest.RequestTimeout)
-		couldRetry := shouldRetry(ctx, response, err)
-		logger.WithContext(ctx).Errorf("failed queryId: %v, couldRetry: %v, retryResponseSuccess: %v, retryErr: %v", qid, couldRetry, retryResponse.Success, retryErr)
 	}
 
 	sc.execRespCache.store(resultPath, response)
 	return response, nil
 }
 
 // we want to retry if the query was not successful, but also did not fail
-func shouldRetry(ctx context.Context, response *execResponse, err error) bool {
-	// if deadline has passed dont retty
-	deadline, ok := ctx.Deadline()
-	if ok && (deadline.Before(deadline)) {
-		return false
-	}
-
-	// if context has been canceled dont retry
-	select {
-	case <-ctx.Done():
-		return false
-	}
-
-	// if there is a response succeeds, dont retry 
-	if response.Success {
-		return false
-	}
-
-	// if there is a response message dont retry 
-	if response.Message != "" {
-		return false
-	}
+func badResponse(ctx context.Context, response *execResponse, qid string, err error) bool {
+	retryable := false
+	// retry if query failed but there is no error 
+	if (!response.Success) && (err == nil) {
```

Comment:
How long do you retry?

Reply:
Like add a retry limit? So far I have seen it always work on one retry so maybe I will limit it to 2 and log if it would need another

Reply:
Ok. Next time let do a standard retry with exponential backoff at most x times under a total budget of xxxx

Reply:
Ok will do!